discord_bots/tasks.py

```diff
@@ -603,3 +610,39 @@ async def vote_passed_waitlist_task():
         ).delete()
         session.delete(vpw)
         session.commit()
+
+
+@tasks.loop(time=config.TRUESKILL_SIGMA_DECAY_JOB_SCHEDULED_TIME)
+async def apply_sigma_decay():
+    session: sqlalchemy.orm.Session
+    with Session() as session:
+        # Lookup all categories to calculate sigma cutoffs
+        # TODO: If we kill off SQLite we can use INTERVAL to check a time range and have a single join between Category and PlayerCategoryTrueskill
+        time_now = datetime.now(timezone.utc)
+        categories = session.query(Category).all()
+        category_details = {
+            category.id: {
+                'category': category,
+                'cutoff': time_now - timedelta(
+                    days=category.sigma_decay_grace_days
+                )
+            }
+            for category in categories
```

Comment:
nit: you might as well filter out categories `where category.sigma_decay_amount != 0`